modules/monitoring-setting-audit-log-levels-for-the-prometheus-adapter.adoc

```diff
@@ -0,0 +1,100 @@
+* You have created the `cluster-monitoring-config` `ConfigMap` object.
+
+.Procedure
+
+You can set an audit log level for the Prometheus Adapter in the default `openshift-monitoring` project:
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add `profile:` under `audit` in the `k8sPrometheusAdapter` section under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    k8sPrometheusAdapter:
+      audit:
+        profile: <audit_log_level> <1>
+----
+<1> The audit log level to apply to the Prometheus Adapter. 
+
+. Set the audit log level by using one of the following values for the `profile:` parameter:
++
+* `None`: Do not log events.
+* `Metadata`: Log only the metadata for the request, such as user, timestamp, and so forth. The request text and the response text will not be logged.
+* `Request`: - Log only the metadata and the request text but not the response text. This option does not apply for non-resource requests.
+* `RequestResponse`: - Log event metadata, request text, and response text. This option does not apply for non-resource requests.
+
+. Save the file to apply the changes. The pods for the Prometheus Adapter restart automatically when you apply the change.
++
+[WARNING]
+====
+When changes are saved to a monitoring config map, the pods and other resources in the related project might be redeployed. The running monitoring processes in that project might also be restarted.
+====
+
+.Verification
+
+. Check that the pods for the Prometheus Adapter are running. The following example lists the status of pods in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring get pods
+----
+
+. Confirm that the correct log level has been applied in the `prometheus-adapter` deployment in the `openshift-monitoring` project. In the following example command, substitute `prometheus-adapter-84484ff64-x99wz` with the pod name for the Prometheus Adapter in your deployment:
++
+[source,terminal]
+----
+$ c -n openshift-monitoring exec -c prometheus-adapter prometheus-adapter-84484ff64-x99wz -- cat /etc/audit/request-profile.yaml
+----
++
+.Example output
+[source,terminal]
+----
+"apiVersion": "audit.k8s.io/v1"
+"kind": "Policy"
+"metadata":
+  "name": "Request"
+"omitStages":
+- "RequestReceived"
+"rules":
+- "level": "Request"
+----
++
+[IMPORTANT]
+====
+If an unrecognized `profile` value is used for the Prometheus Adapter in the `ConfigMap` object, the Prometheus Adapter pods might not restart successfully.
+====
+
+. Review the audit log for the Prometheus Adapter. In the following example command, substitute `prometheus-adapter-84484ff64-x99wz` with the pod name for the Prometheus Adapter in your deployment:
```

Comment:
Review the audit log for the Prometheus Adapter.
this is enough

Reply:
Fixed